prov/gni/include/gnix.h

```diff
@@ -854,6 +859,15 @@ struct gnix_fab_req {
 	char inject_buf[GNIX_INJECT_SIZE];
 };
 
+/*
+ * macro to test whether a request is replayable
+ * or not based on the value of the tx_failures field
+ */
```

Comment:
Also, I'm not strictly opposed to this, but I find it less clear that this has the side effect of incrementing tx_failures.  If you keep it here, please add it to the comments.